fauna/client/client.py

```diff
@@ -279,7 +279,27 @@ def _query(
             if "txn_ts" in response_json:
                 self.set_last_txn_ts(int(response_json["txn_ts"]))
 
-            return QuerySuccess(response_json, headers)
+            stats = response_json["stats"] if "stats" in response_json else None
+            summary = response_json[
+                "summary"] if "summary" in response_json else None
+            query_tags = QueryTags.decode(
+                response_json["query_tags"]
```

Comment:
My read of the spec, and Core's code is that this value comes from the `x-query-tags` header and is a simple string. Are we meant to decode them into a structure across all drivers?

Reply:
Query tags is sent on the headers, but received on the body, still encoded as the k=v string. To me, it's more natural to interact with a map. As the implementation stands in this driver, you provide a map and we put it into the headers in the implementation, and so I think it makes sense to receive a map on the response.

Maybe we should check with core on whether this should be consistent? header out, header in, or body out, body in?

Reply:
> The value of the x-query-tags header, if it was provided.

Oh.. I misread that. Is Core simply passing these back as is? If so, I'm less sure about them being a struct. For instance, DataDog's tags are simple strings that are given some structure if a colon is provided. `mytag,tagkey:tagvalue` are both valid tags. I don't see anything in the spec about what the structure of these tags should be.

Reply:
Yeah, I thought about this, but the spec says single tags without a value are invalid.

Maybe you're right, but I'd still want to give users a class to interact with so they don't have to know about the encoding..